src/fetchers/gh_crate_meta.rs

```diff
@@ -5,22 +5,18 @@ use log::{debug, info, warn};
 use reqwest::Client;
 use reqwest::Method;
 use serde::Serialize;
+use tokio::sync::OnceCell;
 use url::Url;
 
 use super::Data;
-use crate::{download_and_extract, remote_exists, BinstallError, PkgFmt, Template};
+use crate::{
+    download_and_extract, remote_exists, AutoAbortJoinHandle, BinstallError, PkgFmt, Template,
+};
 
 pub struct GhCrateMeta {
     client: Client,
     data: Data,
-}
-
-impl GhCrateMeta {
-    fn url(&self) -> Result<Url, BinstallError> {
-        let ctx = Context::from_data(&self.data);
-        debug!("Using context: {:?}", ctx);
-        ctx.render_url(&self.data.meta.pkg_url)
-    }
+    url: OnceCell<Url>,
```

Comment:
We can simply use `Option` here:

```suggestion
    url: Option<Url>,
```

Reply:
Hmm, I don't think so? We don't have a `&mut self` in the methods.

Reply:
Oh you are right, but we can just use once_cell::sync::OnceCell instead.
The tokio one merely adds async get_or_init and accepts a future, but the code here doesn't use that feature anyway.